e2e/tests/custom-projects/create-custom-project.spec.ts

```diff
@@ -1,52 +1,276 @@
-import { Page, test, expect } from "@playwright/test";
+import { Page, test, expect, Locator } from "@playwright/test";
+import { ACTIVITY } from "@shared/entities/activity.enum";
 import { E2eTestManager } from "@shared/lib/e2e-test-manager";
+import { EXTENDED_TIMEOUT, ROUTES } from "e2e/constants";
 
 let testManager: E2eTestManager;
 let page: Page;
 
 test.describe.configure({ mode: "serial" });
 
-test.describe("Custom Projects", () => {
-  const projectName = "test project";
-  const insertProjectName = async () => {
-    await page.getByRole("textbox", { name: "Insert project name" }).click();
-    await page
-      .getByRole("textbox", { name: "Insert project name" })
-      .fill(projectName);
-  };
-  const submitCustomProjectAndCheckPreview = async () => {
-    await page.getByRole("button", { name: "Continue" }).click();
-    await page.waitForURL("/projects/preview");
-    await expect(
-      page.getByRole("heading", { name: projectName }),
-    ).toBeVisible();
-  };
-
+test.describe("Custom Projects - Create", () => {
   test.beforeAll(async ({ browser }, testInfo) => {
-    testInfo.setTimeout(testInfo.timeout + 60_000);
+    testInfo.setTimeout(testInfo.timeout + EXTENDED_TIMEOUT);
     page = await browser.newPage();
     testManager = await E2eTestManager.load(page);
     await testManager.ingestBaseData();
   });
 
   test.afterEach(async () => {
-    await page.goto("/projects/new");
+    await page.goto(ROUTES.projects.new);
   });
 
   test.afterAll(async () => {
     await testManager.clearDatabase();
     await testManager.close();
   });
 
-  test.describe("Create Custom Projects", () => {
+  test.describe("Form validation", () => {
+    const projectName = "test project";
+    const insertProjectName = async () => {
```

Comment:
This is nice. However, it might potentially be a reusable logic instead of a closure, when (hopefully) the whole test suite gets bigger and bigger?

Not saying it should be, you have more context.

In case it is, 1 option would be to extract the functions (easiest) but then we would need to pass the page as param

Another option would be to abstract it to a helper class that can be injected to the test manager? Or to put it in the test manager? But maybe it is already growing too much and it should only handle only common stuff for all tests cases?